Test failure with pandas 1.3.0: `test_timezone_freq`

I took a look at the codebase and it seems like `pandas.Series.iloc[slice]` used to return a freq and no longer does. The impact on dask is that when using `from_pandas` the freq data is not passed along. @jorisvandenbossche do you know if this is an intentional change in pandas? It doesn't seem too different from other change that you listed in #7100

This is the relevant code: 

```python
    dsk = {
        (name, i): data.iloc[start:stop]
        for i, (start, stop) in enumerate(zip(locations[:-1], locations[1:]))
    }
``` 
https://github.com/dask/dask/blob/c845de789975b405fe17215a7b1d7a9742c92302/dask/dataframe/io/io.py#L230:L233

And this is the failing test:

```python-traceback
=============================================== FAILURES ================================================
_________________________________________ test_timezone_freq[1] _________________________________________

npartitions = 1

    @pytest.mark.parametrize(
        "npartitions",
        [
            1,
            pytest.param(
                2,
                marks=pytest.mark.xfail(
                    not dd._compat.PANDAS_GT_110, reason="Fixed upstream."
                ),
            ),
        ],
    )
    def test_timezone_freq(npartitions):
        s_naive = pd.Series(pd.date_range("20130101", periods=10))
        s_aware = pd.Series(pd.date_range("20130101", periods=10, tz="US/Eastern"))
        pdf = pd.DataFrame({"tz": s_aware, "notz": s_naive})
        ddf = dd.from_pandas(pdf, npartitions=npartitions)
    
>       assert pdf.tz[0].freq == ddf.compute().tz[0].freq == ddf.tz.compute()[0].freq
E       assert <Day> == None
E         +<Day>
E         -None

dask/dataframe/tests/test_dataframe.py:246: AssertionError
_________________________________________ test_timezone_freq[2] _________________________________________

npartitions = 2

    @pytest.mark.parametrize(
        "npartitions",
        [
            1,
            pytest.param(
                2,
                marks=pytest.mark.xfail(
                    not dd._compat.PANDAS_GT_110, reason="Fixed upstream."
                ),
            ),
        ],
    )
    def test_timezone_freq(npartitions):
        s_naive = pd.Series(pd.date_range("20130101", periods=10))
        s_aware = pd.Series(pd.date_range("20130101", periods=10, tz="US/Eastern"))
        pdf = pd.DataFrame({"tz": s_aware, "notz": s_naive})
        ddf = dd.from_pandas(pdf, npartitions=npartitions)
    
>       assert pdf.tz[0].freq == ddf.compute().tz[0].freq == ddf.tz.compute()[0].freq
E       assert <Day> == None
E         +<Day>
E         -None

dask/dataframe/tests/test_dataframe.py:246: AssertionError
```

--- a/dask/dataframe/tests/test_dataframe.py
+++ b/dask/dataframe/tests/test_dataframe.py
@@ -224,9 +224,6 @@ def test_index_names():
     assert ddf.index.compute().name == "x"
 
 
-@pytest.mark.xfail(
-    dd._compat.PANDAS_GT_130, reason="https://github.com/dask/dask/issues/7444"
-)
 @pytest.mark.parametrize(
     "npartitions",
     [

